File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {
    static boolean[] isVisited;
    static StringBuilder sb = new StringBuilder();
    static ArrayList<ArrayList<Integer>> graph = new ArrayList<>();

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] graphInfo = br.readLine().split(" ");
        int vertex = Integer.parseInt(graphInfo[0]);
        int edge = Integer.parseInt(graphInfo[1]);
        int startVertex = Integer.parseInt(graphInfo[2]);
        isVisited = new boolean[vertex + 1];

        for (int i = 0; i <= vertex; i++) {
            graph.add(new ArrayList<>());
        }

        StringTokenizer st;
        for (int i = 0; i < edge; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            graph.get(a).add(b);
            graph.get(b).add(a);
        }

        for (int i = 1; i <= vertex; i++) {
            Collections.sort(graph.get(i));
        }
        dfs(startVertex);
        sb.delete(sb.length()-1, sb.length());
        System.out.println(sb);

        Arrays.fill(isVisited, false);
        sb.setLength(0);
        bfs(startVertex);
        sb.delete(sb.length()-1, sb.length());
        System.out.println(sb);
    }

    public static void dfs(int cur) {
        if (isVisited[cur]) return;

        isVisited[cur] = true;
        sb.append(cur).append(" ");
        for (int i = 0; i < graph.get(cur).size(); i++) {
            dfs(graph.get(cur).get(i));
        }
        return;
    }

    public static void bfs(int start) {
        Queue<Integer> queue = new LinkedList<>();
        queue.add(start);
        isVisited[start] = true;

        while (!queue.isEmpty()) {
            int cur = queue.poll();
            sb.append(cur).append(" ");
            for (int i = 0; i < graph.get(cur).size(); i++) {
                int check = graph.get(cur).get(i);
                if (!isVisited[check]) {
                    isVisited[check] = true;
                    queue.add(check);
                }
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import collections
import sys

vertices, edges, start = map(int, sys.stdin.readline().split())
graph = [[] for _ in range(vertices + 1)]
is_visited = [False] * (vertices + 1)
ans_list = [[] for _ in range(2)]


def dfs(start_num: int) -> None:
    if not is_visited[start_num]:
        is_visited[start_num] = True
        ans_list[0].append(str(start_num))

        for i in range(len(graph[start_num])):
            check_node = graph[start_num][i]

            if not is_visited[check_node]:
                dfs(check_node)


def bfs(start_num: int) -> None:
    queue = collections.deque()
    if not is_visited[start_num]:
        is_visited[start_num] = True
        queue.append(start_num)

    while queue:
        check_num = queue.popleft()
        ans_list[1].append(str(check_num))

        for i in range(len(graph[check_num])):
            check_node = graph[check_num][i]

            if not is_visited[check_node]:
                is_visited[check_node] = True
                queue.append(check_node)


for i in range(edges):
    a, b = map(int, sys.stdin.readline().split())
    graph[a].append(b)
    graph[b].append(a)

is_visited[0] = True
for i in range(len(graph)):
    graph[i] = sorted(graph[i])

dfs(start)
is_visited = [False] * (vertices + 1)
is_visited[0] = True
bfs(start)

sys.stdout.write(' '.join(ans_list[0]) + "\n")
sys.stdout.write(' '.join(ans_list[1]))
